src/Rules/Properties/ReadOnlyPropertyAssignRefRule.php

```diff
@@ -41,7 +41,7 @@ public function processNode(Node $node, Scope $scope): array
 			if (!$scope->canAccessProperty($propertyReflection)) {
 				continue;
 			}
-			if (!$nativeReflection->isReadOnly()) {
+			if (!$nativeReflection->isReadOnly() && !$nativeReflection->isReadOnlyByPhpDoc()) {
```

Comment:
I really wanted a separate set of rules for this. So please copy this rule, it can be named something like `ReadOnlyByPhpDocProperty...` and write a separate testcase for it. And register it only for bleeding edge :)

Reply:
Of course, makes sense. I was not fully sure what you'd prefer.
Regarding bleeding edge I hoped this could be avoided in a minor release. Is the reasoning mostly that already used readonly docs could trigger new annoying errors?

Reply:
It can definitely be annoying and show unexpected errors. For users that use both Psalm and PHPStan, but:

1) Have Psalm errors ignored by the baseline, PHPStan would start reporting new errors about readonly
2) There can be implementation differences between Psalm and PHPStan, so something that's okay for Psalm might be reported by PHPStan

With bleedingEdge we're gonna gather useful feedback by people that willingly early-adopt this stuff 😊